### PR TITLE
[tests-only][full-ci]Added tests for setting password in a public share link

### PR DESCRIPTION
**Pull request by SagarGi**
### Description
This PR adds the tests scenarios to set and update(if already created) the password for the public link share for shareNg
```feature
Scenario: set password for a file's link share
Scenario: update password for a file's link share
```

### Related Issue:
https://github.com/owncloud/ocis/issues/7968

### CODE REVIEW

File: tests/TestHelpers/GraphHelper.php [hunk 1]
```diff
@@ -1701,4 +1701,38 @@ public static function updateLinkShare(
...
+	 * @param string $password
+	 * @param string $spaceId
+	 * @param string $itemId
+	 * @param mixed $body
+	 * @param string $shareId
+	 *
+	 * @return ResponseInterface
+	 * @throws GuzzleException
+	 */
+	public static function setPassword(
```
Comment (saw-jan) [inline tests/TestHelpers/GraphHelper.php:1718]:
```suggestion
	public static function setLinkSharePassword(
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 2]
```diff
@@ -518,3 +518,70 @@ Feature: Create a share link for a resource
         }
       }
       """
+
+  @env-config
+  Scenario: set password for a file's link share
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:523]:
```suggestion
  Scenario: set password on a file's link share
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 3]
```diff
@@ -518,3 +518,70 @@ Feature: Create a share link for a resource
...
+            "type": "boolean",
+            "enum": [true]
+          }
+        }
+      }
+      """
+    And the public should be able to download file "textfile1.txt" from inside the last public link shared folder for shareNg using the new public WebDAV API with password "%public%" and the content should be "other data"
+
+
+  Scenario: update password for a file's link share
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:557]:
```suggestion
  Scenario: update password of a file's link share
```

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php [hunk 4]
```diff
@@ -288,18 +288,24 @@ public function downloadPublicFileInsideAFolderWithRange(string $path, string $r
...
 	 *
 	 * @return ResponseInterface
 	 */
 	public function downloadFileFromPublicFolder(
 		string $path,
 		string $password,
 		string $range,
-		string $publicWebDAVAPIVersion = "old"
+		string $publicWebDAVAPIVersion = "old",
+		bool $downloadForShareNG = false
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/PublicWebDavContext.php:300]: can we utilize `publicWebDAVAPIVersion`?
something like `publicWebDAVAPIVersion="shareNg"`
```suggestion
		string $publicWebDAVAPIVersion = "old"
```

File: tests/acceptance/features/bootstrap/PublicWebDavContext.php [hunk 5]
```diff
@@ -913,6 +919,39 @@ public function shouldBeAbleToDownloadFileInsidePublicSharedFolderWithPasswordAn
...
+		$response = $this->downloadFileFromPublicFolder(
+			$path,
+			$password,
+			"",
+			$publicWebDAVAPIVersion,
+			true
+		);
+		$this->featureContext->checkDownloadedContentMatches($content, "", $response);
+
+		$this->featureContext->theHTTPStatusCodeShouldBe(200, "", $response);
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/PublicWebDavContext.php:952]:
```suggestion
		$this->featureContext->theHTTPStatusCodeShouldBe(200, "", $response);
		$this->featureContext->checkDownloadedContentMatches($content, "", $response);
```

File: tests/acceptance/features/bootstrap/Sharing.php [hunk 6]
```diff
@@ -184,6 +184,18 @@ public function shareNgGetLastCreatedLinkShareID(): string {
 		return $this->getJsonDecodedResponse($lastResponse)['id'];
 	}
 
+	/**
+	 * @return string
+	 */
+	public function shareNGGetLastCreatedPublicLinkShareToken(): string {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/Sharing.php:190]:
```suggestion
	public function shareNgGetLastCreatedLinkShareToken(): string {
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 7]
```diff
@@ -518,3 +518,70 @@ Feature: Create a share link for a resource
...
+        ],
+        "properties": {
+          "hasPassword": {
+            "type": "boolean",
+            "enum": [true]
+          }
+        }
+      }
+      """
+    And the public should be able to download file "textfile1.txt" from inside the last public link shared folder for shareNg using the new public WebDAV API with password "%public%" and the content should be "other data"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:554]:
```suggestion
    And the public should be able to download file "textfile1.txt" from the last link shared folder with password "%public%" using shareNg and the content should be "other data"
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 8]
```diff
@@ -518,3 +518,70 @@ Feature: Create a share link for a resource
...
+    And the public download of file "textfile1.txt" from inside the last public link shared folder for shareNg using the new public WebDAV API with password "$heLlo*1234*" should fail with HTTP status code "401"
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:587]:
```suggestion
    And the public download of file "textfile1.txt" from the last link shared folder with password "$heLlo*1234*" should fail with HTTP status code "401" using shareNg
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature (hunk 7 above)
Comment (saw-jan) [reply tests/acceptance/features/apiSharingNg/linkShare.feature:554]: the step is long. let's reduce it

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 9]
```diff
@@ -518,3 +518,70 @@ Feature: Create a share link for a resource
...
+    Given the following configs have been set:
+      | config                                       | value |
+      | OCIS_SHARING_PUBLIC_SHARE_MUST_HAVE_PASSWORD | false |
+    And user "Alice" has uploaded file with content "other data" to "textfile1.txt"
+    And user "Alice" has created the following link share:
+      | resourceType | file          |
+      | resource     | textfile1.txt |
+      | space        | Personal      |
+      | role         | view          |
+    When user "Alice" sets the following password for the last public link share using the Graph API:
```
Comment (saw-jan) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:533]:
```suggestion
    When user "Alice" sets the following password for the last link share using the Graph API:
```

File: tests/acceptance/features/apiSharingNg/linkShare.feature [hunk 10]
```diff
@@ -518,3 +518,69 @@ Feature: Create a share link for a resource
...
+      | resourceType | file          |
+      | resource     | textfile1.txt |
+      | space        | Personal      |
+      | role         | view          |
+      | password     | $heLlo*1234*  |
+    When user "Alice" updates the following password for the last link share using the Graph API:
+      | resourceType | file          |
+      | resource     | textfile1.txt |
+      | space        | Personal      |
+      | password     | %public%      |
```
Comment (ScharfViktor) [inline tests/acceptance/features/apiSharingNg/linkShare.feature:568]: could you please add extra scenario where the password is from a banned list file?

Comment (SagarGi) [reply tests/acceptance/features/apiSharingNg/linkShare.feature:568]: i will add it to a TODO check list @ScharfViktor. May be not in this PR.
Added : https://github.com/owncloud/ocis/issues/7968

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 11]
```diff
@@ -239,17 +239,59 @@ public function userUpdatesLastPublicLinkShareUsingTheGraphApiWith(string $user,
...
+	 * @When user :user sets/updates the following password for the last link share using the Graph API:
+	 *
+	 * @param string $user
+	 * @param TableNode|null $body
+	 *
+	 * @return void
+	 * @throws Exception
+	 * @throws \GuzzleHttp\Exception\GuzzleException
+	 */
+	public function userSetsOrUpdatesPasswordForPublicLinkShareUsingTheGraphApiWith(string $user, TableNode  $body):void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:265]:
```suggestion
	public function userSetsOrUpdatesFollowingPasswordForLastLinkShareUsingTheGraphApi(string $user, TableNode  $body):void {
```

**Review by saw-jan — COMMENTED** [review]
Others LGTM

**Review by saw-jan — APPROVED** [review]

**Pull request merged**